custom/icds_reports/utils/__init__.py

```diff
@@ -544,3 +546,21 @@ def person_has_aadhaar_column(beta):
 
 def person_is_beneficiary_column(beta):
     return 'cases_person_beneficiary_v2'
+
+
+def track_time(func):
+    """A decorator to track the duration an aggregation script takes to execute"""
+    from custom.icds_reports.models import AggregateSQLProfile
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        start = time.time()
+        result = func(*args, **kwargs)
+        end = time.time()
+        AggregateSQLProfile(
+            name=str(func),
+            duration=int(end - start)
```

Comment:
postgres/django has a date time range field that might have been good for this https://docs.djangoproject.com/en/2.0/ref/contrib/postgres/fields/#datetimerangefield